### PR TITLE
[12.0][IMP] sales_team_security: Contacts permissions revamp

### PR DESCRIPTION
**Pull request by pedrobaeza**
Restrict contacts by code, removing the need of manually manipulating the existing record rules, that can also have some side effects.

@Tecnativa TT24290

### CODE REVIEW

**Review by sergio-teruel — APPROVED** [review]
:+1: Code review... ok form me. Waiting travis

**Review by HaraldPanten — APPROVED** [review]
Seems to be OK 👍

File: sales_team_security/models/ir_rule.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    def _compute_domain(self, model_name, mode="read"):
+        """Inject extra domain for restricting partners when the user
+        has the group 'Sales / User: Own Documents Only'.
+        """
+        res = super()._compute_domain(model_name, mode=mode)
+        user = self.env.user
+        group1 = "sales_team.group_sale_salesman"
+        group2 = "sales_team_security.group_sale_team_manager"
+        group3 = "sales_team.group_sale_salesman_all_leads"
+        if model_name == "res.partner":
```
Comment (yajo) [inline sales_team_security/models/ir_rule.py]: :thinking: If you're affecting only the `res.partner` model, don't you think `res.partner.check_access_rule()` would be the appropriate hook to override?

Comment (pedrobaeza) [reply sales_team_security/models/ir_rule.py]: That method only serves for direct operations over the record, but for example it doesn't limit the number or records returned by a `search`. But this low level method is call for all of them.

Comment (yajo) [reply sales_team_security/models/ir_rule.py]: Interesting, today I learned that.

**Review by yajo — APPROVED** [review]

**Pull request merged**